docs/book/mlops-stacks/data-validators/great-expectations.md

```diff
@@ -0,0 +1,456 @@
+Great Expectations works with tabular data in a variety of formats and data
+sources, of which ZenML currently supports only `pandas.DataFrame` as part of
+its pipelines.
+
+You should use the Great Expectations Data Validator when you need the following
+data validation features that are possible with Great Expectations:
+
+* [Data Profiling](https://docs.greatexpectations.io/docs/guides/expectations/how_to_create_and_edit_expectations_with_a_profiler):
+generates a set of validation rules (Expectations) automatically by inferring
+them from the properties of an input dataset.
+
+* [Data Quality](https://docs.greatexpectations.io/docs/guides/validation/checkpoints/how_to_pass_an_in_memory_dataframe_to_a_checkpoint):
+runs a set of predefined or inferred validation rules (Expectations) against an
+in-memory dataset.
+
+* [Data Docs](https://docs.greatexpectations.io/docs/terms/data_docs): generate
+and maintain human readable documentation of all your data validation rules,
+data quality checks and their results.
+
+You should consider one of the other [Data Validator flavors](./data-validators.md#data-validator-flavors)
+if you need a different set of data validation features.
+
+## How do you deploy it?
+
+The Great Expectations Data Validator flavor is included in the Great
+Expectations ZenML integration, you need to install it on your local machine to
+be able to register a Great Expectations Data Validator and add it to your stack:
+
+```shell
+zenml integration install great_expectations -y
+```
+
+Depending on how you configure the Great Expectations Data Validator, it can
+reduce or even completely eliminate the complexity associated with setting up
+the store backends for Great Expectations. If you're only looking for a quick
+and easy way of adding Great Expectations to your stack and are not concerned
+with the configuration details, you can simply run:
+
+```shell
+# Register the Great Expectations data validator
+zenml data-validator register ge_data_validator --flavor=great_expectations
+
+# Register and set a stack with the new data validator
+zenml stack register custom_stack -dv ge_data_validator ... --set
+```
+
+{% tabs %}
+{% tab title="ZenML Managed Great Expectations (default)" %}
+
+The default Data Validator setup plugs Great Expectations directly into the
+[Artifact Store](../artifact-stores/artifact-stores.md) component that is part
+of the same stack. As a result, the Expectation Suites, Validation Results and
+Data Docs are stored in the ZenML Artifact Store and you don't have to configure
+Great Expectations at all, ZenML takes care of that for you:
+
+```shell
+# Register the Great Expectations data validator
+zenml data-validator register ge_data_validator --flavor=great_expectations
+
+# Register and set a stack with the new data validator
+zenml stack register custom_stack -dv ge_data_validator ... --set
+```
+
+{% hint style="warning" %}
```

Comment:
It's great you put those trade-offs here, but right now it sounds a bit like you're screwed no matter which option you choose 😄  I'd suggest to remove the warnings and instead have a short paragraph before where you explain when to use what "- If you are ... do A, if you want to ... do B, ..."

Reply:
I had a feeling my approach was too bleak :) Thanks for your suggestion, I will try to sweeten it a bit.

Reply:
done